components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/ApplicationCreationWSWorkflowExecutor.java

```diff
@@ -43,119 +43,92 @@
+                            + "        <wor:applicationCallbackUrl>$3</wor:applicationCallbackUrl>\n"
+                            + "        <wor:applicationDescription>$4</wor:applicationDescription>\n"
+                            + "        <wor:tenantDomain>$5</wor:tenantDomain>\n"
+                            + "        <wor:userName>$6</wor:userName>\n"
+                            + "        <wor:workflowExternalRef>$7</wor:workflowExternalRef>\n"
+                            + "        <wor:callBackURL>$8</wor:callBackURL>\n"
+                            + "      </wor:ApplicationApprovalWorkFlowProcessRequest>";
+
+            ApplicationWorkflowDTO appWorkFlowDTO = (ApplicationWorkflowDTO) workflowDTO;
+            Application application = appWorkFlowDTO.getApplication();
+            String callBackURL = appWorkFlowDTO.getCallbackUrl();
+
+
+            payload = payload.replace("$1", application.getName());
+            payload = payload.replace("$2", application.getTier());
+            payload = payload.replace("$3", application.getCallbackUrl());
+            payload = payload.replace("$4", application.getDescription());
+            payload = payload.replace("$5", appWorkFlowDTO.getTenantDomain());
+            payload = payload.replace("$6", appWorkFlowDTO.getUserName());
+            payload = payload.replace("$7", appWorkFlowDTO.getExternalWorkflowReference());
+            payload = payload.replace("$8", callBackURL != null ? callBackURL : "?");
+
+            client.fireAndForget(AXIOMUtil.stringToOM(payload));
+
+        } catch (AxisFault axisFault) {
+            log.error("Error sending out message", axisFault);
+            throw new WorkflowException("Error sending out message", axisFault);
+        } catch (XMLStreamException e) {
+            log.error("Error converting String to OMElement", e);
+            throw new WorkflowException("Error converting String to OMElement", e);
+        }
+
+    }
+
+    /**
+     * Complete the external process status.
+     * Based on the workflow , we will update the status column of the
+     * Application table
+     *
+     * @param workFlowDTO
+     */
+    @Override
+    public void complete(WorkflowDTO workFlowDTO) throws WorkflowException {
         workFlowDTO.setUpdatedTime(System.currentTimeMillis());
         ApiMgtDAO dao = new ApiMgtDAO();
         try {
-            if(dao.getApplicationById(Integer.parseInt(workFlowDTO.getWorkflowReference())) != null) {
+            if (dao.getApplicationById(Integer.parseInt(workFlowDTO.getWorkflowReference())) != null) {
 
                 super.complete(workFlowDTO);
-                log.info("Application Creation [Complete] Workflow Invoked. Workflow ID : " + workFlowDTO.getExternalWorkflowReference() + "Workflow State : " + workFlowDTO.getStatus());
+                log.info("Application Creation [Complete] Workflow Invoked. Workflow ID : " + workFlowDTO

```

Comment:
Used info other than debug because Application creation is not a frequent operation and as a mechanism to identify that a workflow has been executed.